BadHabbit/BadHabbit/Kit/NetworkKit/RequestExecutor.swift

```diff
@@ -30,13 +30,27 @@ class RequestExecutor: RequestExecutorProtocol {
     func perform<Parser: ResponseParserProtocol>(
         request: Request,
         parser: Parser,
-        completion: (Result<Parser.Response, Error>) -> Void
+        completion: @escaping (Result<Parser.Response, Error>) -> Void
     ) {
-        do {
-            let response = try parser.parse(Data())
-            completion(.success(response))
-        } catch {
-            completion(.failure(error))
+        if let url = URL(string: request.path) {
+            let urlSession = URLSession.shared.dataTask(with: url) {(data, response, error) in
+                if let error = error {
+                    completion(.failure(error))
```

Comment:
In case if both error and data aren't nil completion will be called twice. This isn't correct